internal/pkg/iostreams/io.go

```diff
@@ -156,7 +161,46 @@ func (s *system) ReadSecret() ([]byte, error) {
 		return nil, fmt.Errorf("prompting is disabled")
 	}
 
-	return term.ReadPassword(int(s.in.Fd()))
+	fd := int(s.in.Fd())
+
+	// Store and restore the terminal status on interruptions to
+	// avoid that the terminal remains in the password state
+	// This is necessary as for https://github.com/golang/go/issues/31180
+	oldState, err := term.GetState(fd)
+	if err != nil {
+		return make([]byte, 0), err
+	}
+
+	type Buffer struct {
+		Buffer []byte
+		Error  error
+	}
+	errorChannel := make(chan Buffer, 1)
+
+	// SIGINT and SIGTERM restore the terminal, otherwise the no-echo mode would remain intact
+	interruptChannel := make(chan os.Signal, 1)
+	signal.Notify(interruptChannel, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
So the context will already be cancelled from those signals: https://github.com/hashicorp/hcp/blob/main/main.go#L42

So instead, just listen to the context being cancelled.

Reply:
Thank you. Copy 🍝 got the better of me